File: C++/Pelicula.cpp
#include <cstring>
#include <iostream>
#include "Pelicual.h"

using namespace std;

Pelicula::Pelicula() {
	strcpy(this->titulo, "");
	strcpy(this->director, "");
	this->duracion = 0;
	this->numeroActores = 0;
}

Pelicula::Pelicula(char titulo[], char director[], int duracion) {
	strcpy(this->titulo, titulo);
	strcpy(this->director, director);
	this->duracion = duracion;
	this->numeroActores = 0;
}

char* Pelicula::getTitulo() {
	return this->titulo;
}

void Pelicula::setTitulo(char nuevo_titulo[]) {
	strcpy(this->titulo, nuevo_titulo);
}

char* Pelicula::getDirector() {
	return this->director;
}

void Pelicula::setDirector(char nuevo_director[]) {
	strcpy(this->director, nuevo_director);
}

int Pelicula::getDuracion() {
	return this->duracion;
}

void Pelicula::setDuracion(int nueva_duracion) {
	this->duracion = nueva_duracion;
};

void Pelicula::mostrarPelicula() {
	cout<<"El nombre de la pelicula es "<<this->getTitulo()<<endl;
	cout<<"Su director es "<<this->getDirector()<<endl;
	cout<<"la duracion de la pelicula es "<<this->getDuracion()<<endl;
	for(int i = 0;i<this->numeroActores;i++){
		cout<<"En la peli trabaja: "<<endl;
		this->actores[i].mostrarActor();
	}
};

bool Pelicula::isActor(Actor actor) {
	bool encontrado = false;
	for(int i=0;i < numeroActores; i++) {
		if(strcmp(actor.getNombre(), this->actores[i].getNombre()) == 0&& actor.getNacimiento() == this->actores[i].getNacimiento){
			encontrado = true;
		}
	}
	return encontrado;
};

void Pelicula::introduceActor(Actor actor) {
	if(this->numeroActores < 15) {
		this->actores[this->numeroActores] = actor;
		this->numeroActores++;
	}
};

File: JAVA/peliculas/src/main/java/Pelicula.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author Ricardo Mendez
 */
public class Pelicula {
    private String titulo;
    private String director;
    private int duracion;
    
    //Atributos que van a representar la relacion de asociacion
    private Actor[] actores = new Actor[15];
    private int numeroActores;
    
    public Pelicula() {
        this.titulo = "";
        this.director = "";
        this.duracion = 0;
        this.numeroActores = 0;
    }
    
    public Pelicula(String titulo, String director, int duracion) {
        this.titulo = titulo;
        this.director = director;
        this.duracion = duracion;
    }
    
    public String getTitulo() {
        return this.titulo;
    }
    
    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }
    
    public String getDirector() {
        return this.director;
    }
    
    public void setDirector(String director) {
        this.director = director;
    }
    
    public int getDuracion() {
        return this.duracion;
    }
    
    public void SetDuracion(int duracion) {
        this.duracion = duracion;
    }
    
    public void mostrarPelicula() {
        System.out.println("El nombre de la pelicula es " + this.getTitulo());
        System.out.println("El director es " + this.getDirector());
        System.out.println("La duracion de la pelicula es " + this.getDuracion());
        for(int i=0;i<this.numeroActores;i++) {
            System.out.println("En la pelicula trabaja ");
            this.actores[i].mostrarActor();
        }
    }
    
    public boolean isActor(Actor actor) {
        boolean encontrado = false;
        for(int i=0;i<this.numeroActores;i++) {
            if(actor.getNombre() == this.actores[i].getNombre() && actor.getNacimiento() == this.actores[i].getNacimiento()) {
                encontrado = true;
            }
        }
        return encontrado;
    }
    
    public void addActor(Actor actor) {
        if(this.numeroActores < 15){
            this.actores[this.numeroActores] = actor;
            this.numeroActores ++;
        }
    }
    
}
